src/pages/Governance/components/TransactionResponseSnackbar.tsx

```diff
@@ -0,0 +1,222 @@
+import * as RRD from "react-router-dom";
+import React from "react";
+import {Types} from "aptos";
+import {
+  Snackbar,
+  Alert,
+  Box,
+  Button,
+  IconButton,
+  Typography,
+  Link,
+} from "@mui/material";
+import CloseIcon from "@mui/icons-material/Close";
+import RefreshIcon from "@mui/icons-material/Refresh";
+import {
+  TransactionResponse,
+  TransactionResponseOnError,
+  TransactionResponseOnSubmission,
+} from "../../../api/hooks/useSubmitTransaction";
+import {useGetTransaction} from "../../../api/hooks/useGetTransaction";
+
+function CloseAction({onCloseSnackbar}: {onCloseSnackbar: () => void}) {
+  return (
+    <IconButton
+      size="small"
+      aria-label="close"
+      color="inherit"
+      onClick={onCloseSnackbar}
+    >
+      <CloseIcon fontSize="small" />
+    </IconButton>
+  );
+}
+
+function RefreshAction() {
+  const refreshPage = () => {
+    window.location.reload();
+  };
+
+  return (
+    <Box alignSelf="center" marginRight={1.5}>
+      <Button
+        variant="outlined"
+        color="inherit"
+        size="large"
+        onClick={refreshPage}
+        endIcon={<RefreshIcon />}
+      >
+        Refresh
+      </Button>
+    </Box>
+  );
+}
+
+type SuccessSnackbarProps = {
+  transactionHash: string;
+  refreshOnSuccess?: boolean;
+  onCloseSnackbar: () => void;
+};
+
+function SuccessSnackbar({
```

Comment:
might be nice to have success and failure response in 2 separate files for maintainability and readability. they can all be imported and exported from this file though